### PR TITLE
Add MarkdownLint CLI checker

### PR DESCRIPTION
**Pull request by dholm**
Adds support for the MarkdownLint Command Line Interface based on the
Node.js markdownlint package.

 - https://github.com/igorshubovych/markdownlint-cli

### CODE REVIEW

**Comment by dholm** [comment]
@cpitclaudel Done!

**Comment by cpitclaudel** [comment]
Thanks! One last thing: you should use `.. syntax-checker-config-file` instead of `.. defcustom` for the config file (that's what causing the build to fail)

**Comment by dholm** [comment]
Oops, fixed!

File: CHANGES.rst [hunk 1]
```diff
@@ -1,5 +1,6 @@
-32-cvs (in development)
-=======================
+=========================
+ 32-cvs (in development)
+=========================
```
Comment (fmdkdd) [inline CHANGES.rst]: I'm not sure it was necessary to restyle every headings.  But I guess it's your mode that made the change sneakily?

Comment (dholm) [reply CHANGES.rst]: Hmm, yes, I didn't even catch that myself.

Comment (dholm) [reply CHANGES.rst]: Fixed

**Comment by fmdkdd** [comment]
Great!  We just need you to squash your commits before we can merge :)

**Comment by dholm** [comment]
@fmdkdd Memo to self, never edit code on GitHub. :/

**Comment by fmdkdd** [comment]
@dholm Thanks.  One last thing: how does `markdown-cli` compare to `mdl`, the other linter we already have in Flycheck?  I want to know if it's worth changing the default to `markdown-cli`, or whether we should keep `mdl` (because it has more lints/has better lints/is more performant/more popular or whatever).

**Comment by dholm** [comment]
@fmdkdd I used mdl at first but had issues with suppressions not working according to the documentation, for instance disabling the line-length check for tables.  markdownlint-cli appears to be better maintained and did not have these issues.

**Comment by fmdkdd** [comment]
@dholm I see, thank you.

@marsam You added `mdl`, do you have an opinion on whether we should switch to `markdownlint-cli` as the default checker for Markdown?

**Comment by cpitclaudel** [comment]
Ping @dholm , any opinions on changing the default checker? :)

**Comment by marsam** [comment]
Sorry for the late response, I don't have any issues with changing `markdownlint-cli` as default checker

**Comment by cpitclaudel** [comment]
@marsam Merged, thanks @dholm! (And sorry for the delay)  Let's change the default checker in a separate PR. @dholm, would you like to open one?

**Comment by dholm** [comment]
@cpitclaudel Thank you! Isn't it already the default though? I thought the order in `flycheck-checkers` is what makes it default.

**Pull request merged**